src/transaction/log_applier_sql_log.c

```diff
@@ -642,6 +686,10 @@ create_dir (const char *new_dir)
 	{
 	  if (mkdir (path, 0777) < 0)
 	    {
+	      er_stack_push ();
```

Comment:
Is it for only printing the error without any side effects? Can't we just er_set() and return the error instead of ER_FAILED below?

Reply:
Yes, that reason. But I don't completely understand the er_set() mechanism. I just follow the following codes and check that the error message is well written in the error log. I don't want to overwrite the previous error code.
```
8031   if (prm_get_bool_value (PRM_ID_HA_SQL_LOGGING))
8032     {
8033       if (sl_init (la_slave_db_name, log_path) != NO_ERROR)
8034         {
8035           er_stack_push ();
8036           er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_HA_GENERIC_ERROR, 1, "Failed to initialize SQL logger");
8037           er_stack_pop ();
8038         }
8039       else
8040         {
8041           la_enable_sql_logging = true;
8042         }
8043     }
```